specs/bellatrix/fork-choice.md

```diff
@@ -78,6 +79,100 @@ As per EIP-3675, before a post-transition block is finalized, `notify_forkchoice
 The `safe_block_hash` parameter MUST be set to return value of
 [`get_safe_execution_payload_hash(store: Store)`](../../fork_choice/safe-block.md#get_safe_execution_payload_hash) function.
 
+##### `should_override_forkchoice_update`
+
+If proposer boost re-orgs are implemented and enabled (see `get_proposer_head`) then additional care
+must be taken to ensure that the proposer is able to build an execution payload.
+
+If a beacon node knows it will propose the next block then it SHOULD NOT call
+`notify_forkchoice_updated` if it detects the current head to be weak and potentially capable of
+being re-orged. Complete information for evaluating `get_proposer_head` _will not_ be available
+immediately after the receipt of a new block, so an approximation of those conditions should be
+used when deciding whether to send or suppress a fork choice notification. The exact conditions
+used may be implementation-specific, a suggested implementation is below.
```

Comment:
Thanks for this! It makes sense
It may not be clear to readers at first, but I don't think there's a good reason to call `should_override_forkchoice_update` if the block `N` is timely. (Assume you are a proposer for `N+1`). If the block is timely, then you should just call `notify_forkchoice_updated`

Reply:
Yeah I think you're right that should be the first check. I made a related suggestion about the ordering here: https://github.com/ethereum/consensus-specs/pull/3034#discussion_r999983626

Reply:
Verygood👌🙏🌹

Reply:
👌🌹🙏